### PR TITLE
Various fixes.

### PR DESCRIPTION
**Pull request by alesstimec**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1403/
Test PASSed.

File: internal/v5/auth.go [hunk 1]
```diff
@@ -110,43 +110,13 @@ func (h *ReqHandler) authorizeEntityAndTerms(req *http.Request, entityIds []*rou
...
-			}
-		}
+	public, acls, requiredTerms, err := h.entityAuthInfo(entityIds)
+	if err != nil {
+		return authorization{}, errgo.Mask(err)
 	}
+
 	// if all entities are open to everyone and non of the entities defines any Terms, then we return nil
-	if allowAuth {
+	if public {
```
Comment (cmars) [inline internal/v5/auth.go:119]: A publicly available charm with terms should still require agreement to them. Shouldn't this be `if public && len(requiredTerms) == 0`?

Comment (alesstimec) [inline internal/v5/auth.go:119]: If any entity requires terms public will be set to false

Comment (cmars) [inline internal/v5/auth.go:119]: Ah, I see now. Thanks.

**Comment by cmars** [comment]
LGTM. Let's land this and update the PR into v5-unstable.

**Comment by cmars** [comment]
Just tested manually btw and looks good!

**Pull request merged**